irnl_rdt_correction/input_options.py

```diff
@@ -0,0 +1,219 @@
+"""
+Input Options
+-------------
+
+Handles the input parameters, contains their default values
+and checks for their validity.
+"""
+import argparse
+import logging
+from typing import Iterable, Sized, Union
+
+from irnl_rdt_correction.constants import EXT_TFS, EXT_MADX
+from irnl_rdt_correction.equation_system import SOLVER_MAP
+from irnl_rdt_correction.utilities import list2str, DotDict
+
+LOG = logging.getLogger(__name__)
+
+
+# Default Values ---------------------------------------------------------------
+
+DEFAULTS = {'feeddown': 0,
+            'ips': [1, 2, 5, 8],
+            'accel': 'lhc',
+            'solver': 'lstsq',
+            'update_optics': True,
+            'iterations': 1,
+            'ignore_corrector_settings': False,
+            'rdts2': None,
+            'ignore_missing_columns': False,
+            'output': None,
+            }
+
+DEFAULT_RDTS = {
+    'lhc': ('F0003', 'F0003*',  # correct a3 errors with F0003
+            'F1002', 'F1002*',  # correct b3 errors with F1002
+            'F1003', 'F3001',  # correct a4 errors with F1003 and F3001
+            'F4000', 'F0004',  # correct b4 errors with F4000 and F0004
+            'F6000', 'F0006',  # correct b6 errors with F6000 and F0006
+            ),
+    'hllhc': ('F0003', 'F0003*',  # correct a3 errors with F0003
+              'F1002', 'F1002*',  # correct b3 errors with F1002
+              'F1003', 'F3001',  # correct a4 errors with F1003 and F3001
+              'F0004', 'F4000',  # correct b4 errors with F0004 and F4000
+              'F0005', 'F0005*',  # correct a5 errors with F0005
+              'F5000', 'F5000*',  # correct b5 errors with F5000
+              'F5001', 'F1005',  # correct a6 errors with F5001 and F1005
+              'F6000', 'F0006',  # correct b6 errors with F6000 and F0006
+              ),
+}
+
+
+# Parser -----------------------------------------------------------------------
+
+def get_parser() -> argparse.ArgumentParser:
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "--twiss",
+        dest="twiss",
+        nargs="+",
+        help="Path(s) to twiss file(s). Defines which elements to correct for!",
```

Comment:
> Defines which elements to correct for!

Not sure I fully understand what you mean by that ... is it meant as `Should contain the elements with errors and correctors `? Is there a harm in providing a full LHC twiss?

Reply:
If that's what you want to correct for, no. But if you only want to correct for elements in the IRs you need to give it a twiss with the elements from the IRs.